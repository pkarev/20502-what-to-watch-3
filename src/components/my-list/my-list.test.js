import React from 'react';
import renderer from 'react-test-renderer';
import MyList from './my-list.jsx';

const movies = [
  {
    id: 1,
    name: `Fantastic Beasts`,
    previewImage: `/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    previewImage: `/img/bohemian-rhapsody.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 3,
    name: `Moonrise kindom`,
    previewImage: `/img/moonrise-kingdom.jpg`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  }
];

it(`Render MyList`, () => {
  const tree = renderer
    .create(
        <MyList movies={movies} onCardClick={() => {}}/>, {
          createNodeMock: () => ({})
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
