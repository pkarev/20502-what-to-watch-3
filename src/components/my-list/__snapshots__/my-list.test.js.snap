// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyList 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          poster="/img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        >
          <source
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </video>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Fantastic Beasts
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          poster="/img/bohemian-rhapsody.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        >
          <source
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </video>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <video
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          poster="/img/moonrise-kingdom.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        >
          <source
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </video>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Moonrise kindom
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
